spring-context/src/test/java/org/springframework/cache/CacheReproTests.java

```diff
@@ -131,7 +132,7 @@ public void spr13081ConfigFailIfCacheResolverReturnsNullCacheName() {
 	}
 
 	@Test
-	public void spr14230AdaptsToOptional() {
+	public void spr14230AdaptsToOptionalAndResolvesCacheWithExpression() {
```

Comment:
I think maybe there's no need to rigorously test this since it's using the already existing logic of `CacheOperationExpressionEvaluator`.

Here's the logs from this run (not the **Computed cache name**):
```java
12:43:04.885 [Test worker] TRACE o.s.c.a.AnnotationCacheOperationSource - Adding cacheable method 'insertItem' with attribute: [Builder[public org.springframework.beans.testfixture.beans.TestBean org.springframework.cache.CacheReproTests$Spr14230Service.insertItem(org.springframework.beans.testfixture.beans.TestBean)] caches=[itemCache-1] | key='#item.name' | keyGenerator='' | cacheManager='' | cacheResolver='' | condition='' | unless='']
12:43:04.902 [Test worker] TRACE o.s.c.a.AnnotationCacheOperationSource - Adding cacheable method 'findById' with attribute: [Builder[public java.util.Optional org.springframework.cache.CacheReproTests$Spr14230Service.findById(java.lang.String)] caches=[itemCache-#{'1'}] | key='' | keyGenerator='' | cacheManager='' | cacheResolver='' | condition='' | unless='' | sync='false']
12:43:04.954 [Test worker] TRACE o.s.c.i.CacheInterceptor - Computed cache key 'tb1' for operation Builder[public org.springframework.beans.testfixture.beans.TestBean org.springframework.cache.CacheReproTests$Spr14230Service.insertItem(org.springframework.beans.testfixture.beans.TestBean)] caches=[itemCache-1] | key='#item.name' | keyGenerator='' | cacheManager='' | cacheResolver='' | condition='' | unless=''
12:43:04.974 [Test worker] TRACE o.s.c.i.SimpleCacheResolver - Computed cache name 'itemCache-1' for operation Builder[public java.util.Optional org.springframework.cache.CacheReproTests$Spr14230Service.findById(java.lang.String)] caches=[itemCache-#{'1'}] | key='' | keyGenerator='' | cacheManager='' | cacheResolver='' | condition='' | unless='' | sync='false'
12:43:04.974 [Test worker] TRACE o.s.c.i.CacheInterceptor - Computed cache key 'tb1' for operation Builder[public java.util.Optional org.springframework.cache.CacheReproTests$Spr14230Service.findById(java.lang.String)] caches=[itemCache-#{'1'}] | key='' | keyGenerator='' | cacheManager='' | cacheResolver='' | condition='' | unless='' | sync='false'
12:43:04.975 [Test worker] TRACE o.s.c.i.CacheInterceptor - Cache entry for key 'tb1' found in cache 'itemCache-1'
12:43:05.026 [Test worker] TRACE o.s.c.i.SimpleCacheResolver - Computed cache name 'itemCache-1' for operation Builder[public java.util.Optional org.springframework.cache.CacheReproTests$Spr14230Service.findById(java.lang.String)] caches=[itemCache-#{'1'}] | key='' | keyGenerator='' | cacheManager='' | cacheResolver='' | condition='' | unless='' | sync='false'
12:43:05.026 [Test worker] TRACE o.s.c.i.CacheInterceptor - Computed cache key 'tb1' for operation Builder[public java.util.Optional org.springframework.cache.CacheReproTests$Spr14230Service.findById(java.lang.String)] caches=[itemCache-#{'1'}] | key='' | keyGenerator='' | cacheManager='' | cacheResolver='' | condition='' | unless='' | sync='false'
12:43:05.027 [Test worker] TRACE o.s.c.i.CacheInterceptor - No cache entry for key 'tb1' in cache(s) [itemCache-1]
12:43:05.027 [Test worker] TRACE o.s.c.i.CacheInterceptor - Computed cache key 'tb1' for operation Builder[public java.util.Optional org.springframework.cache.CacheReproTests$Spr14230Service.findById(java.lang.String)] caches=[itemCache-#{'1'}] | key='' | keyGenerator='' | cacheManager='' | cacheResolver='' | condition='' | unless='' | sync='false'

```